internal/api/schema.go

```diff
@@ -167,3 +169,71 @@ func (a *API) DeleteVersion(ctx context.Context, in *stencilv1beta1.DeleteVersio
 		Message: message,
 	}, err
 }
+
+func (a *API) DetectSchemaChange(writer http.ResponseWriter, request *http.Request, pathParams map[string]string) error {
+	namespaceID := pathParams["namespaceId"]
+	schemaID := pathParams["schemaId"]
+
+	fromVersion := request.URL.Query().Get("from")
+	toVersion := request.URL.Query().Get("to")
+
+	versionList, err := a.schema.ListVersions(context.Background(), namespaceID, schemaID)
+	if err != nil {
+		return fmt.Errorf("error getting version list - %s", err.Error())
+	}
+	if len(versionList) == 0 {
+		return fmt.Errorf("got empty version list")
+	}
+	if len(versionList) == 1 {
+		return fmt.Errorf("only one version exists for schema %s", schemaID)
+	}
+	latestVersion := versionList[len(versionList)-1]
+
+	if fromVersion == "" {
+		fromVersion = strconv.Itoa(int(latestVersion - 1))
+	}
+	if toVersion == "" {
+		toVersion = strconv.Itoa(int(latestVersion))
+	}
+
+	fromVer, errFrom := strconv.ParseInt(fromVersion, 10, 32)
```

Comment:
Once you got the request values from the url/query-param, before proceeding further, validate immediately. Here, before calling ListVersion, these two variables needs to be validated.

If the from or to is empty string "", then use -1 for fromVer/toVer a convention that the value is not set. if it is not empty string then parse int. 
Also error checking of those should be immediately after you execute it. i.e. after line 199, line202-204 then line 200 followed by 205-207.

Reply:
Also by doing this you can use `err` variable throughout the flow (in 99.9% sceanrios) where the error check would've already happened.

i.e. line 211 & 215 can have `err` variable only.